mono/metadata/assembly.c

```diff
@@ -4662,14 +4671,21 @@ mono_assembly_request_byname (MonoAssemblyName *aname, const MonoAssemblyByNameR
 {
 	MonoDomain *domain = mono_domain_get ();
 	mono_trace (G_LOG_LEVEL_DEBUG, MONO_TRACE_ASSEMBLY, "Request to load %s in (domain %p, alc %p)", aname->name, domain, (gpointer)req->request.alc);
-	MonoAssembly *result = mono_assembly_request_byname_nosearch (aname, req, status);
+	MonoAssembly *result;
+	if (status)
+		*status = MONO_IMAGE_OK;
+#ifndef ENABLE_NETCORE
+	result = mono_assembly_request_byname_nosearch (aname, req, status);
 	const gboolean refonly = req->request.asmctx == MONO_ASMCTX_REFONLY;
 
 	if (!result && !req->no_postload_search) {
 		/* Try a postload search hook */
 		result = mono_assembly_invoke_search_hook_internal (req->request.alc, req->requesting_assembly, aname, refonly, TRUE);
 		result = prevent_reference_assembly_from_running (result, refonly);
 	}
+#else
+	result = netcore_load_reference (aname, req->request.alc, req->requesting_assembly, !req->no_postload_search);
```

Comment:
What do you think about getting rid of `netcore_load_reference` entirely and replacing its other callers by a call to `mono_assembly_request_byname`?

Reply:
It's definitely an option. I'm not super happy with what the call stack looks like right now (since it's somewhat convoluted) but I also don't really want to require building a bunch more requests for simple calls within assembly.c, so I've been trying to think on how I want to organize things.